cxx-qt-gen/src/parser/qobject.rs

```diff
@@ -0,0 +1,479 @@
+// SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use crate::syntax::{
+    attribute::{attribute_find_path, attribute_tokens_to_ident},
+    path::{path_angled_args_to_type_path, path_compare_str, path_to_single_ident},
+};
+use quote::format_ident;
+use std::collections::HashMap;
+use syn::{
+    spanned::Spanned, Error, Ident, ImplItem, Item, ItemEnum, ItemImpl, ItemStruct, Result, Type,
+    TypePath,
+};
+
+#[derive(Default)]
+pub struct ParserCxxQtData {
+    /// Map of the QObjects defined in the module that need generation
+    pub qobjects: HashMap<Ident, ParserQObject>,
+    /// Any use blocks end up in the CXX-Qt generated module
+    pub uses: Vec<Item>,
+}
+
+/// A representation of a QObject within a CXX-Qt ItemMod
+///
+/// This has initial splitting of Item's into relevant blocks, other phases will
+/// then mutate these Item's for generation purposes.
+#[derive(Default)]
+pub struct ParserQObject {
```

Comment:
```suggestion
pub struct ParsedQObject {
```

This feels a bit more intuitive to me, but I don't have a strong opinion on this either way.

Reply:
Makes sense i think, keep the main `Parse::from(...)` but then the inner things are `ParsedT`